packages/shared/comms/v4/InstanceConnection.ts

```diff
@@ -82,6 +84,22 @@ export class InstanceConnection implements RoomConnection {
         }
         const transport = createTransport(config, islandChangedMessage)
 
+        if (this.peerLeftListener) {
+          await this.bff.removeSystemTopicListener(this.peerLeftListener)
+        }
+        this.peerLeftListener = await this.bff.addSystemTopicListener(
+          `island.${islandChangedMessage.islandId}.peer_left`,
+          (data: Uint8Array) => {
+            try {
+              const peerLeftMessage = LeftIslandMessage.decode(Reader.create(data))
+              removePeerByUUID(peerLeftMessage.peerId)
```

Comment:
this should be a message of the transport, only called from the handlers.ts file